docs/tpm2-pkcs11_object_auth_model.md

```diff
@@ -0,0 +1,528 @@
+# tpm2-pkcs11 object & authorization model
+
+## Introduction
+
+The objective of this document is to present an architectural overview of the
+tpm2-pkcs11 object and authorization model implemented in the design. The topics
+covered here in are:
+
+1. [PKCS11 Objects](#pkcs11-objects)
+2. [PKCS11 Roles](#pkcs11-roles)
+3. [TPM2 Objects](#tpm2-objects)
+4. [tpm2-pkcs11 Objects](#tpm2-pkcs11-objects)
+5. [NV Object Diagram](#nv-object-diagram)
+6. [NV Policy Diagram](#nv-policy-diagram)
+7. [Transient Object Diagram](#transient-object-diagram)
+8. [Transient Policy Diagram](#transient-policy-diagram)
+9. [Sample tpm2-tools implementation](#sample-tpm2-tools-implementation)
+10. [Important note on authorization](#important-note-on-authorization)
+
+## PKCS11 Objects
+An Overview of the hierarchical concept of objects.
+<p>
+  <img src="illustrations/reader-slot-token-obj.png"
+</p>
+<br>
+<br>
+
+* The PKCS11 READER is a medium through with information is exchanged with the
+PKCS11 device or TOKEN.
+* Every READER has a SLOT that may contain a TOKEN.
+* The TOKEN ultimately protects all the USER created objects as well as access
+through an authorization model to distinguish the SO, USER or adversary.
+* USER objects can be keys, certificates or any user defined data.
+
+## PKCS11 Roles
+Overview of pkcs11 requirements of the security officer and user roles. The
+PKCS11 design describes two specific roles - A security-officer or **SO**
+and **USER**. These roles are realized with separate PIN authentications.
+Following are some relevant design considerations when comparing the roles:
+* Neither SO nor USER can know each other's PIN.
+* The SO is restricted to an administrative role to modify SO PIN and or an
+existing USER's PIN.
+* While both the **SO and the USER can change USER PIN**, only a SO can change
+SO PIN.
+* USER authentication PIN is a must for operating on USER objects.
+* The SO cannot access any USER objects.
+* While SO could change USER PIN, the **SO cannot restore the old USER PIN**.
+This intentionally prevents SO from having explicit access to USER objects and
+also to bring traceability to security officer actions regarding PIN changes for
+USER authorization.
+<br>
+As an example, a token initialized with a SO PIN can be setup with a default
+USER PIN. The USER then changes the PIN to his liking which the SO cannot know
+& thus cant restore. The USER can now use the new PIN to create and operate on
+USER objects. If the USER is convinced of forgetting the PIN, USER can get the
```

Comment:
USER forgets his pin. I don't know why the USER was convinced to forget his pin, did you break their legs or something?